### PR TITLE
core/local/chokidar: Extract steps

### PR DESCRIPTION
**Pull request by sebn**
- Makes `ChokidarWatcher` smaller
- Makes it easier to understand/document steps
- Better matches the `AtomWatcher` design

---

- [x] PR is not too big
- [x] it improves UX & DX in some way
- [x] it includes unit tests matching the implementation changes
- [x] it includes scenarios matching a new behaviour or has been manually tested
- [x] it includes relevant documentation

### CODE REVIEW

File: core/local/chokidar/send_to_prep.js [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+
+type SendToPrepOpts = {
+  pouch: Pouch,
+  prep: Prep
+}
+
+*/
+
+const log = logger({
+  component: 'ChokidarWatcher'
```
Comment (sebn) [inline core/local/chokidar/send_to_prep.js]: `component: 'chokidar/send_to_prep'`

File: core/local/chokidar/send_to_prep.js [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+          break
+        case 'FileUpdate':
+          await onChange(c, prep)
+          break
+        case 'FileAddition':
+          await onAddFile(c, prep)
+          break
+        case 'FileMove':
+          await onMoveFile(c, prep)
+          if (c.update) await onChange(c.update, prep)
```
Comment (sebn) [inline core/local/chokidar/send_to_prep.js]: @taratatach Yesterday bug was here: I was passing `c` instead of `c.update` 🤦‍♂

**Comment by sebn** [comment]
Renamed logger + done with manual testing. LGTM. Documentation will come up in a future PR (it didn't exist anyway).

File: core/local/chokidar/send_to_prep.js (hunk 2 above)
Comment (taratatach) [reply core/local/chokidar/send_to_prep.js]: Good catch tests! :+1:

File: core/local/chokidar/initial_scan.js [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+  const docs = await pouch.byRecursivePathAsync('')
+  const inInitialScan = doc =>
+    initialScan.ids.indexOf(metadata.id(doc.path)) !== -1
+
+  // the Syncdir is empty error only occurs if there was some docs beforehand
+  let emptySyncDir = docs.length > NB_OF_DELETABLE_ELEMENT
+  let unappliedMoves = []
+
+  for (const doc of docs) {
+    if (inInitialScan(doc) || doc.trashed || doc.incompatibilities) {
```
Comment (taratatach) [inline core/local/chokidar/initial_scan.js]: Do we need to check for incompatibilities in the Atom Watcher as well? Or maybe it's not even necessary here?

**Pull request merged**